pkg/engine/virtualmachinescalesets.go

```diff
@@ -98,6 +99,21 @@ func CreateMasterVMSS(cs *api.ContainerService) VirtualMachineScaleSetARM {
 		virtualMachine.Identity = identity
 	}
 
+	for _, addonProfile := range addonProfiles {
+		if addonProfile.Enabled && addonProfile.Identity != nil && addonProfile.Identity.ResourceID != "" {
+			if virtualMachine.Identity == nil {
+				virtualMachine.Identity = &compute.VirtualMachineScaleSetIdentity{}
```

Comment:
Is there a possibility to move the Identity handling to a utility function and reuse it in VMA and VMSS case ?

Reply:
Thanks for reminding, it makes sense, updated.